client/src/components/Auth/AuthForm.tsx

```diff
@@ -1,121 +1,184 @@
+    value: password,
+    isValid: passwordIsValid,
+    hasError: passwordHasError,
+    valueChangeHandler: passwordChangeHandler,
+    valueBlurHandler: passwordBlurHandler,
+    reset: passwordResetHandler,
+  }: IUserInput = useInput((value: string) => value.trim() !== "");
 
 
   const [isLogin, setIsLogin] = useState(true);
   const [isLoading, setIsLoading] = useState(false);
+  const [list, setList] =  useState<IToastProps[] | []>([]);
+
 
   const switchAuthModeHandler = () => {
     setIsLogin((prevState) => !prevState);
   };
 
+
+  let formIsValid = false;
+  if (isLogin){
+    if(emailIsValid && passwordIsValid) formIsValid = true;
+  }else{
+    if(nameIsValid && emailIsValid && passwordIsValid) formIsValid = true;
+  }
+
+  let toastProperties: IToastProps ;
+
+  const showToastHandler : (type:string, message:string) => void = (type,message) =>{
+    switch(type) {
+      case 'success':
+        toastProperties = {
+          id: list.length+1,
+          title: 'Success',
+          description: message,
+          backgroundColor: '#5cb85c'
+        }
+        break;
+      case 'fail':
+        toastProperties = {
+          id: list.length+1,
+          title: 'Error',
+          description: message,
+          backgroundColor: '#d9534f'
+        }
+        break;
+      default:
+        break ;
+    }
+    setList([...list, toastProperties]);
+  }
+
   const submitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
-    const email = emailInputRef.current?.value;
-    const password = passwordInputRef.current?.value;
-    const name = nameInputRef.current?.value;
-    console.log(email,name,password);
-    // optional: Add validation
+    // Add validation
+    if(!formIsValid){
+      return;
+    }
 
     setIsLoading(true);
     let url;
     if (isLogin) {
-      url =
-        'http://localhost:9000/user/login';
+      url = "http://localhost:9000/user/login";
     } else {
-      url =
-        'http://localhost:9000/user/register';
+      url = "http://localhost:9000/user/register";
     }
-    try 
-    {
-        const response = await Axios({
-            method: 'POST',
-            url: url,
-            data: {
-                name,
-                email,
-                password
-            }
-        });
-
-        if (response.status === 200)
-        {
-            // Could also use history.push('/login')
-            setIsLoading(false);
-            dispatch(authActions.login(response.data))
-            navigate('/');
-        }
-        else
-        {   
-            setIsLoading(false);
-            console.log("unspecified");
-        }
-    } 
-    catch (error) 
-    {
-        setIsLoading(false);
-        const errObj = error as AxiosError
-        const err = errObj.response?.data;
-        const response = err as AxiosError;
-        const notify = () => toast.error(response.message);
-        notify();
+    try {
+      const response = await Axios({
```

Comment:
for data fetching make separate services for each resource like users, todos etc and call the service methods in components

Reply:
I've included a fetch custom hook for user sign in/sign up and then using the function returned by custom hook in AuthForm component.